src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Delete/DeletionService.cs

```diff
@@ -446,12 +456,14 @@ await Parallel.ForEachAsync(includedResources, cancellationToken, async (item, i
                 {
                     await _retryPolicy.ExecuteAsync(async () => await fhirDataStore.Value.HardDeleteAsync(new ResourceKey(item.Resource.ResourceTypeName, item.Resource.ResourceId), request.DeleteOperation == DeleteOperation.PurgeHistory, request.AllowPartialSuccess, innerCt));
                     parallelBag.Add((item.Resource.ResourceTypeName, item.Resource.ResourceId, item.SearchEntryMode == ValueSets.SearchEntryMode.Include));
+                    AppendDeletedSearchParameter(deletedSearchParameters, item.Resource);
```

Comment:
Since this is still deleting the search parameters in the BulkDeleteProcessingJob after the delete has already happened there is still a risk of failing to delete the SPs if the job is shut down.
Could the SPs be removed here instead? Change the AppendDeletedSearchParameter method to call the SearchParameterOperator instead of making a list for the processing job to use.
If this is possible I think it would also be better to have the SP be deleted before the resource is deleted, that way if the job is interrupted and retried the resource still exists to be searched for and deleted.